File: codegen/results/v002/m3-max-48gb/ollama/gemma4_31b-mlx_think/20260910_102409/typescript/002_word_frequency/solution.ts
import * as fs from 'fs';
import * as path from 'path';

/**
 * This program reads a text file, counts the frequency of each word,
 * and outputs the result sorted by frequency descending and alphabetically ascending.
 */
function solve() {
    const inputPath = path.join('input', 'text.txt');

    try {
        // Read the file content
        const content = fs.readFileSync(inputPath, 'utf8');

        // Process the text:
        // 1. Convert to lowercase
        // 2. Replace all non-letter characters with spaces to treat them as word boundaries
        // 3. Split by whitespace and filter out empty strings
        const words = content
            .toLowerCase()
            .replace(/[^a-z\s]/g, ' ')
            .split(/\s+/)
            .filter(word => word.length > 0);

        // Map to store word frequencies
        const wordCounts = new Map<string, number>();
        for (const word of words) {
            wordCounts.set(word, (wordCounts.get(word) || 0) + 1);
        }

        // Convert Map to an array of [word, count] pairs for sorting
        const sortedWords = Array.from(wordCounts.entries()).sort((a, b) => {
            const [wordA, countA] = a;
            const [wordB, countB] = b;

            // Sort by count descending
            if (countB !== countA) {
                return countB - countA;
            }
            // For ties, sort alphabetically ascending
            return wordA.localeCompare(wordB);
        });

        // Print each unique word and its count in the format "word: count"
        for (const [word, count] of sortedWords) {
            process.stdout.write(`${word}: ${count}\n`);
        }
    } catch (err) {
        // Handle file not found or read errors silently as per prompt "Do not output anything else"
        // however, in a real app, we would log this.
    }
}

// Execute the solve function
solve();

File: codegen/results/v002/m3-max-48gb/ollama/gemma4_31b-mlx_think/20260910_102409/go/002_word_frequency/solution.go
package main

import (
	"fmt"
	"os"
	"regexp"
	"sort"
	"strings"
)

type wordCount struct {
	word  string
	count int
}

func main() {
	// Read the file
	content, err := os.ReadFile("input/text.txt")
	if err != nil {
		return
	}

	// Convert to lowercase
	text := strings.ToLower(string(content))

	// Remove all characters that are not letters or whitespace
	// We replace non-letters (except spaces/newlines) with spaces to avoid merging words
	reg := regexp.MustCompile(`[^a-z\s]`)
	text = reg.ReplaceAllString(text, " ")

	// Split by whitespace
	words := strings.Fields(text)

	// Count frequencies
	countsMap := make(map[string]int)
	for _, w := range words {
		countsMap[w]++
	}

	// Convert map to slice for sorting
	var sortedCounts []wordCount
	for w, c := range countsMap {
		sortedCounts = append(sortedCounts, wordCount{w, c})
	}

	// Sort by count descending, then by word ascending
	sort.Slice(sortedCounts, func(i, j int) bool {
		if sortedCounts[i].count != sortedCounts[j].count {
			return sortedCounts[i].count > sortedCounts[j].count
		}
		return sortedCounts[i].word < sortedCounts[j].word
	})

	// Output results
	for _, wc := range sortedCounts {
		fmt.Printf("%s: %d\n", wc.word, wc.count)
	}
}